content/blog/embrace-kubernetes-part1/index.md

```diff
@@ -0,0 +1,96 @@
+
+#### Do you have the people?
+
+Kubernetes will demand a team of people who are dedicated to its implementation and support. The implementation itself is quite complex. The Kubernetes architecture involves what is basically a distributed database called etcd, and there are additional layers on top of it. A robust implementation demands a deep understanding of that multi-layer architecture. This complexity is somewhat mitigated by the managed services many cloud providers now offer, but it’s still important to understand what’s going on.
+
+Secondly, the actual interface required to deploy something with Kubernetes is also complex. It can be many thousands of lines of YAML, which often needs to be templated and manipulated with mechanisms that YAML was never designed to use. Take a look at [Why the **** are we templating YAML?](https://leebriggs.co.uk/blog/2019/02/07/why-are-we-templating-yaml.html) for a strong opinion on this matter.
+
+Finally, when something goes wrong, you need a subject matter expert to fix it. Because Kubernetes gives you such a high degree of abstraction, it’s very difficult to debug and most people will be unfamiliar with what’s going on under the hood. You need someone who’s really knowledgeable when there’s trouble. In fact, you probably need two people. You’re too vulnerable if that person leaves the company and you’re completely unreasonable if you can’t even let that person take a vacation.
+
+#### Can you afford the technical debt?
+
+Because you’re dedicating several people to Kubernetes, you may be increasing your technical debt in other areas. Do you still have people available to handle bugs and other support issues?
+
+#### Can you afford the extra infrastructure?
+
+As with any large migration, moving from your current system to Kubernetes needs to happen cautiously. You’re not simply going to destroy your existing infrastructure. Instead, you’ll probably want to adopt a side-by-side approach and slowly move your applications across. In other words, for quite a while, you’ll need to support two infrastructures rather than one.
+
+### Is Kubernetes More than You Need?
+
+Many companies, and startups in particular, don’t want to plan for their current user base. They want an infrastructure, such as Kubernetes, that they think will support the million users they see in their future. Does this sort of planning make sense?
+
+Perhaps it’s better to spend your time understanding what your business value is and building up that value. For infrastructure, when you’re starting out, consider off-the-shelf options such as Heroku or serverless compute.
+
+Architectures have to be shaped over time. You can't simply create your infrastructure once and think you’re finished.Whatever infrastructure you adopt, you need to be able to change and adapt. In five years time, who knows what will be available? Don’t lock yourself into an architecture prematurely.
+
+Learn More
```

Comment:
This seems to repeat the next paragraph?

Reply:
deleted